<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsAttributesFormInitCode</class>
 <widget class="QDialog" name="QgsAttributesFormInitCode">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="2" column="0">
    <widget class="QgsCodeEditorPython" name="mInitCodeEditorPython" native="true"/>
   </item>
   <item row="3" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QWidget" name="mInitCodeSourceContainter" native="true">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>40</height>
      </size>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="toolTip">
         <string>QGIS forms can have a Python function that is called when the form is opened.
Use this function to add extra logic to your forms.
The function code of the function can be loaded from the source code entered 
in this dialog,  from an external python file or from the environment (for example 
from a plugin or from startup.py).

An example is:

          from PyQt4.QtGui import QWidget

          def my_form_open(dialog, layer, feature):
	geom = feature.geometry()
	control = dialog.findChild(QWidget,&quot;MyLineEdit&quot;)

Reference in function name: my_form_open

</string>
        </property>
        <property name="text">
         <string>Python Init function</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="mInitCodeSourceComboBox">
        <property name="toolTip">
         <string>QGIS forms can have a Python function that is called when the form is opened.
Use this function to add extra logic to your forms.
The function code of the function can be loaded from the source code entered 
in this dialog,  from an external python file or from the environment (for example 
from a plugin or from startup.py).

An example is:

          from PyQt4.QtGui import QWidget

          def my_form_open(dialog, layer, feature):
	geom = feature.geometry()
	control = dialog.findChild(QWidget,&quot;MyLineEdit&quot;)

Reference in function name: my_form_open

</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QWidget" name="mInitFunctionContainer" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>40</height>
      </size>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="mInitFunctionLabel">
        <property name="text">
         <string>Function name</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="mInitFunctionLineEdit">
        <property name="toolTip">
         <string>Enter the name of the form init function.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="mInitFilePathLabel">
        <property name="text">
         <string>External file</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="mInitFilePathLineEdit"/>
      </item>
      <item>
       <widget class="QToolButton" name="pbtnSelectInitFilePath">
        <property name="text">
         <string>â€¦</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCodeEditorPython</class>
   <extends>QWidget</extends>
   <header>qgscodeeditorpython.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>QgsAttributesFormInitCode</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsAttributesFormInitCode</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
